libraries/CMakeLists.txt

```diff
@@ -2,15 +2,33 @@
 add_subdirectory(${3RDPARTY_DIR})
 
 # Macro utility to clone the coreMQTT submodule.
-macro( clone_mqtt )
+macro( clone_library submodule_dir )
```

Comment:
What is the reason of submoduling each spoke repository separately instead of submoduling all libraries as once with `git submodule update --init --recursive -- libraries/`?

Reply:
It has the added advantage of not having to do a recursive clone and only taking what is needed for each submodule's submodules.